src/components/summary/SummaryComponent.tsx

```diff
@@ -132,13 +137,27 @@ export function SummaryComponent({ summaryNode, overrides }: ISummaryComponent)
           [classes.border]: !display?.hideBottomBorder,
         })}
       >
-        <SummaryComponentSwitch
-          summaryNode={summaryNode}
-          targetNode={targetNode}
-          change={change}
-          label={label}
-          overrides={overrides}
-        />
+        {RenderSummary && component instanceof FormComponent ? (
```

Comment:
Yup, but typescript doesn't agree with that.. 🤷 It complains because it didn't infer that component must be a FormComponent if RenderSummary is set, so the call to `component.renderSummaryBoilerplate()` below fails without that check.